spec/page/posts_spec.rb

```diff
@@ -1,24 +1,44 @@
 require 'spec_helper'
 
 RSpec.describe 'Fb::Page#posts' do
-  context 'given an invalid access token' do
-    let(:page) { Fb::Page.new access_token: 'invalid_token' }
-
-    it 'raises Fb::HTTPError' do
-      expect{page.posts}.to raise_error Fb::HTTPError
-    end
-  end
-
-  context 'given a valid access_token' do
-    let(:user) { Fb::User.new access_token: ENV['FB_TEST_ACCESS_TOKEN'] }
-    let(:page) { user.pages.first }
+  let(:user) { Fb::User.new access_token: ENV['FB_TEST_ACCESS_TOKEN'] }
+  let(:page) { user.pages.first }
 
+  context 'given a valid access token' do
     it 'returns array of Post for any given page' do
       expect(page.posts).to be_a(Array)
       expect(page.posts).to all (be_a Fb::Post)
       expect(page.posts.map &:id).to all(be_a String)
       expect(page.posts.map &:type).to all(be_a String)
+      expect(page.posts.map &:url).to all(be_a String)
+      expect(page.posts.map &:type).to all(be_a String)
       expect(page.posts.map &:created_at).to all(be_a Time)
+      expect(page.posts.map &:length).to all(be_a String)
+    end
+
+    it 'returns post length as n/a for nonvideos' do
+      none_videos = page.posts.select{|post| post.type != 'video'}
+      expect(none_videos.map &:length).to all(eq 'n/a')
+    end
```

Comment:
@S1v4 Instead, you can add this sort of tests to `spec/post/length_spec.rb`.

In that way, you can make two custom Post with different options and verify that length is either correct or 'n/a'.

That would also fulfill your code coverage to: https://coveralls.io/builds/12769855/source?filename=lib%2Ffb%2Fpost.rb#L40